### PR TITLE
Refresh plugin for August 2023

### PR DESCRIPTION
**Pull request by basil**
Update this plugin to conform to the latest best practices for Jenkins plugins from the Jenkins archetype, including building/testing on Java 17/21.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,16 +1,15 @@
-# Intellij files
-#IDE
-.idea/
-**/*.iml
+target
```
Comment (basil) [inline .gitignore:1]: The recommended `.gitignore` file for Jenkins plugins from the Jenkins archetype.

File: .gitignore [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
 
-# jenkins stuff
-work/
+# mvn hpi:run
+work
 
-# maven stuff
-.mvn/maven.config
```
Comment (basil) [inline .gitignore:10]: This file needs to be checked in for [Incrementals](https://github.com/jenkinsci/incrementals-tools).

File: .mvn/extensions.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<extensions xmlns="http://maven.apache.org/EXTENSIONS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/EXTENSIONS/1.0.0 http://maven.apache.org/xsd/core-extensions-1.0.0.xsd">
```
Comment (basil) [inline .mvn/extensions.xml:1]: See https://github.com/jenkinsci/incrementals-tools.

File: .mvn/maven.config [hunk 4]
```diff
@@ -0,0 +1,2 @@
+-Pconsume-incrementals
```
Comment (basil) [inline .mvn/maven.config:1]: See https://github.com/jenkinsci/incrementals-tools.

File: Jenkinsfile [hunk 5]
```diff
@@ -1,2 +1,5 @@
 // Builds a module using https://github.com/jenkins-infra/pipeline-library
-buildPlugin(platforms: ['linux'])
\ No newline at end of file
+buildPlugin(useContainerAgent: true, configurations: [
+  [platform: 'linux', jdk: 21],
+  [platform: 'windows', jdk: 17],
+])
```
Comment (basil) [inline Jenkinsfile:5]: Switch from building on Java 8 to building on Java 17 and 21, per the recommended configuration in the archetype.

File: pom.xml [hunk 6]
```diff
@@ -4,38 +4,32 @@
     <parent>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>plugin</artifactId>
-        <version>4.33</version>
+        <version>4.72</version>
```
Comment (basil) [inline pom.xml:7]: Latest at the time of this writing.

File: pom.xml [hunk 7]
```diff
@@ -4,38 +4,32 @@
...
         <relativePath />
     </parent>
     <groupId>com.atlassian.jira.software.cloud.jenkins</groupId>
     <artifactId>atlassian-jira-software-cloud</artifactId>
-    <version>2.0.8-SNAPSHOT</version>
+    <version>${revision}${changelist}</version>
```
Comment (basil) [inline pom.xml:12]: See https://github.com/jenkinsci/incrementals-tools.

File: pom.xml [hunk 8]
```diff
@@ -4,38 +4,32 @@
...
     <packaging>hpi</packaging>
     <properties>
+        <revision>2.0.8</revision>
+        <changelist>-SNAPSHOT</changelist>
```
Comment (basil) [inline pom.xml:16]: See https://github.com/jenkinsci/incrementals-tools.

File: pom.xml [hunk 9]
```diff
@@ -4,38 +4,32 @@
...
         <!-- Baseline Jenkins version you use to build the plugin. Users must have this version or newer to run. -->
-        <jenkins.version>2.249.3</jenkins.version>
-        <jenkins.baseline>2.249</jenkins.baseline>
-        <jenkins-test-harness.version>2.49</jenkins-test-harness.version>
-        <java.level>8</java.level>
-        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <jenkins.version>2.387.3</jenkins.version>
+        <jenkins.baseline>2.387</jenkins.baseline>
```
Comment (basil) [inline pom.xml:19]: See [the documentation](https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/).

File: pom.xml [hunk 10]
```diff
@@ -4,38 +4,32 @@
...
-    <version>2.0.8-SNAPSHOT</version>
+    <version>${revision}${changelist}</version>
     <packaging>hpi</packaging>
     <properties>
+        <revision>2.0.8</revision>
+        <changelist>-SNAPSHOT</changelist>
         <!-- Baseline Jenkins version you use to build the plugin. Users must have this version or newer to run. -->
-        <jenkins.version>2.249.3</jenkins.version>
-        <jenkins.baseline>2.249</jenkins.baseline>
-        <jenkins-test-harness.version>2.49</jenkins-test-harness.version>
```
Comment (basil) [inline pom.xml:18]: Defined in the parent POM to a suitably recent version already.

File: pom.xml [hunk 11]
```diff
@@ -4,38 +4,32 @@
...
-        <java.level>8</java.level>
```
Comment (basil) [inline pom.xml:19]: Long deprecated and removed from the plugin parent POM.

File: pom.xml [hunk 12]
```diff
@@ -4,38 +4,32 @@
...
-        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (basil) [inline pom.xml:20]: Moved to the plugin parent POM some time ago.

File: pom.xml [hunk 13]
```diff
@@ -4,38 +4,32 @@
...
+        <jenkins.version>2.387.3</jenkins.version>
+        <jenkins.baseline>2.387</jenkins.baseline>
+        <gitHubRepo>jenkinsci/atlassian-jira-software-cloud-plugin</gitHubRepo>
         <filterModified>true</filterModified>
-        <mockito.version>4.3.1</mockito.version>
```
Comment (basil) [inline pom.xml:22]: Defined in the plugin parent POM now.

File: pom.xml [hunk 14]
```diff
@@ -4,38 +4,32 @@
...
-        <jackson.version>2.13.1</jackson.version>
-        <jackson-databind.version>2.13.1</jackson-databind.version>
```
Comment (basil) [inline pom.xml:24]: Defined in the plugin BOM.

File: pom.xml [hunk 15]
```diff
@@ -4,38 +4,32 @@
...
         <okhttp.version>4.0.1</okhttp.version>
-        <assertj.version>3.11.1</assertj.version>
+        <assertj.version>3.24.2</assertj.version>
         <maven-checkstyle-plugin.version>2.17</maven-checkstyle-plugin.version>
         <archunit.version>0.9.2</archunit.version>
-        <jsonassert.version>1.5.0</jsonassert.version>
+        <jsonassert.version>1.5.1</jsonassert.version>
         <resilience4j.version>1.4.0</resilience4j.version>
-        <slf4j.version>1.7.30</slf4j.version>
```
Comment (basil) [inline pom.xml:31]: Provided by Jenkins core.

File: pom.xml [hunk 16]
```diff
@@ -4,38 +4,32 @@
...
-        <maven.compiler.target>1.8</maven.compiler.target>
-        <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (basil) [inline pom.xml:33]: Defined in the plugin parent POM.

File: pom.xml [hunk 17]
```diff
@@ -4,38 +4,32 @@
...
         <jwt.version>3.18.3</jwt.version>
     </properties>
     <name>Atlassian Jira Software Cloud</name>
     <description>Atlassian Jira Software Cloud Jenkins Integration</description>
-    <url>https://github.com/jenkinsci/atlassian-jira-software-cloud-plugin</url>
+    <url>https://github.com/jenkinsci/${project.artifactId}-plugin</url>
```
Comment (basil) [inline pom.xml:32]: Per the archetype.

File: pom.xml [hunk 18]
```diff
@@ -65,10 +59,10 @@
...
 
     <scm>
-        <connection>scm:git:https://github.com/jenkinsci/atlassian-jira-software-cloud-plugin</connection>
-        <developerConnection>scm:git:https://github.com/jenkinsci/atlassian-jira-software-cloud-plugin</developerConnection>
-        <url>https://github.com/jenkinsci/atlassian-jira-software-cloud-plugin</url>
-        <tag>atlassian-jira-software-cloud-2.0.0</tag>
+        <connection>scm:git:https://github.com/${gitHubRepo}.git</connection>
+        <developerConnection>scm:git:git@github.com:${gitHubRepo}.git</developerConnection>
+        <url>https://github.com/${gitHubRepo}</url>
+        <tag>${scmTag}</tag>
```
Comment (basil) [inline pom.xml:65]: Per the archetype.

File: pom.xml [hunk 19]
```diff
@@ -81,7 +75,6 @@
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>junit</artifactId>
 
-            <scope>compile</scope>
```
Comment (basil) [inline pom.xml:84]: Unnecessary scope.

File: pom.xml [hunk 20]
```diff
@@ -103,39 +96,6 @@
...
-                <exclusion>
-                    <groupId>org.slf4j</groupId>
-                    <artifactId>log4j-over-slf4j</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>org.slf4j</groupId>
-                    <artifactId>jcl-over-slf4j</artifactId>
-                </exclusion>
-            </exclusions>
-        </dependency>
```
Comment (basil) [inline pom.xml:119]: Duplicated what was already in the plugin parent POM.

File: pom.xml [hunk 21]
```diff
@@ -103,39 +96,6 @@
...
-
-        <!-- https://mvnrepository.com/artifact/org.slf4j/slf4j-jdk14 -->
-        <dependency>
-            <groupId>org.slf4j</groupId>
-            <artifactId>slf4j-jdk14</artifactId>
-            <version>1.7.26</version>
-            <scope>test</scope>
-        </dependency>
```
Comment (basil) [inline pom.xml:127]: Already provided by Jenkins core.

File: pom.xml [hunk 22]
```diff
@@ -103,39 +96,6 @@
...
-        <!--  <dependency>
-        <groupId>org.jenkins-ci.main</groupId>
-            <artifactId>jenkins-war</artifactId>
-            <exclusions>
-            <exclusion>
-            <groupId>org.slf4j</groupId>
-            <artifactId>slf4j-jdk14</artifactId>
-            </exclusion>
-            </exclusions>
-        </dependency> -->
```
Comment (basil) [inline pom.xml:138]: Dead code.

File: pom.xml [hunk 23]
```diff
@@ -156,35 +116,11 @@
...
-            <artifactId>slf4j-api</artifactId>
-            <version>${slf4j.version}</version>
-            <!-- <exclusions>
-             <exclusion>
-             <groupId>org.slf4j</groupId>
-             <artifactId>slf4j-jdk14</artifactId>
-             </exclusion>
-             </exclusions> -->
-
-        </dependency>
```
Comment (basil) [inline pom.xml:173]: Already provided by Jenkins core.

File: pom.xml [hunk 24]
```diff
@@ -156,35 +116,11 @@
...
-        <dependency>
-            <groupId>com.fasterxml.jackson.core</groupId>
-            <artifactId>jackson-core</artifactId>
-            <version>${jackson.version}</version>
-        </dependency>
-        <dependency>
-            <groupId>com.fasterxml.jackson.core</groupId>
-            <artifactId>jackson-databind</artifactId>
-            <version>${jackson-databind.version}</version>
             <exclusions>
```
Comment (basil) [inline pom.xml:119]: Already provided by the Jackson 2 API Jenkins plugin.

File: pom.xml [hunk 25]
```diff
@@ -202,20 +138,14 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>jackson2-api</artifactId>
-            <exclusions>
-                <exclusion>
-                    <groupId>com.fasterxml.jackson.core</groupId>
-                    <artifactId>jackson-core</artifactId>
-                </exclusion>
-            </exclusions>
```
Comment (basil) [inline pom.xml:210]: No need for this exclusion and in fact it was breaking the build when the above duplicate dependency was removed.

File: pom.xml [hunk 26]
```diff
@@ -202,20 +138,14 @@
...
         </dependency>
 
 
         <!-- Test dependencies -->
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-aggregator</artifactId>
-            <version>2.6</version>
+            <version>596.v8c21c963d92d</version>
```
Comment (basil) [inline pom.xml:148]: Latest version at the time of this writing.

File: pom.xml [hunk 27]
```diff
@@ -241,7 +171,6 @@
         <dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-core</artifactId>
-            <version>${mockito.version}</version>
```
Comment (basil) [inline pom.xml:244]: Defined in the plugin parent POM.

File: pom.xml [hunk 28]
```diff
@@ -282,7 +218,7 @@
             <dependency>
                 <groupId>io.jenkins.tools.bom</groupId>
                 <artifactId>bom-${jenkins.baseline}.x</artifactId>
-                <version>841.vd6e713d848ab</version>
+                <version>2357.v1043f8578392</version>
```
Comment (basil) [inline pom.xml:221]: Latest at the time of this writing.

File: pom.xml [hunk 29]
```diff
@@ -335,26 +271,6 @@
...
-                    <!-- attached to Maven test phase -->
-                    <execution>
-                        <id>report</id>
-                        <phase>test</phase>
-                        <goals>
-                            <goal>report</goal>
-                        </goals>
-                    </execution>
-                </executions>
-            </plugin>
```
Comment (basil) [inline pom.xml:357]: Defined in the plugin parent POM's `enable-jacoco` profile.

**Review by bgvozdev — APPROVED** [review]
Thanks @basil for contributing!

**Pull request merged**